packages/flutter_tools/lib/src/commands/build_ios_framework.dart

```diff
@@ -0,0 +1,386 @@
+    await bundleBuilder.build(
+      platform: TargetPlatform.ios,
+      buildMode: mode,
+      // Relative paths show noise in the compiler https://github.com/dart-lang/sdk/issues/37978.
+      mainPath: fs.path.absolute(targetFile),
+      assetDirPath: destinationAppFrameworkDirectory.childDirectory('flutter_assets').path,
+      precompiledSnapshot: mode != BuildMode.debug,
+    );
+    status.stop();
+  }
+
+  Future<void> _produceAotAppFrameworkIfNeeded(BuildMode mode, Directory iPhoneModeDirectory, Directory destinationAppFrameworkDirectory) async {
+    if (mode == BuildMode.debug) {
+      return;
+    }
+    final Status status = logger.startProgress(' ├─Building Dart AOT for App.framework...', timeout: timeoutConfiguration.slowOperation);
+    await aotBuilder.build(
+      platform: TargetPlatform.ios,
+      outputPath: iPhoneModeDirectory.path,
+      buildMode: mode,
+      // Relative paths show noise in the compiler https://github.com/dart-lang/sdk/issues/37978.
+      mainDartFile: fs.path.absolute(targetFile),
+      quiet: true,
+      reportTimings: false,
+      iosBuildArchs: <DarwinArch>[DarwinArch.armv7, DarwinArch.arm64],
+    );
+
+    const String appFrameworkName = 'App.framework';
+    copyDirectorySync(iPhoneModeDirectory.childDirectory(appFrameworkName), destinationAppFrameworkDirectory);
+    status.stop();
+  }
+
+  Future<void> _producePlugins(String xcodeBuildConfiguration, Directory iPhoneModeDirectory, Directory iPhoneSimulatorDirectory, Directory modeDirectory,  Directory outputDirectory) async {
+    final Status status = logger.startProgress(' ├─Building plugins...', timeout: timeoutConfiguration.slowOperation);
+    List<String> pluginsBuildCommand = <String>[
+      'xcrun',
+      'xcodebuild',
+      '-alltargets',
+      '-sdk',
+      'iphoneos',
+      '-configuration',
+      xcodeBuildConfiguration,
+      'SYMROOT=${iPhoneModeDirectory.path}',
+      'ONLY_ACTIVE_ARCH=NO' // No device targeted, so build all valid architectures.
+    ];
+
+    final FlutterProject flutterProject = FlutterProject.current();
+    await processUtils.run(
+      pluginsBuildCommand,
+      workingDirectory: flutterProject.ios.hostAppRoot.childDirectory('Pods').path,
+      allowReentrantFlutter: false,
+    );
+
+    pluginsBuildCommand = <String>[
+      'xcrun',
+      'xcodebuild',
+      '-alltargets',
+      '-sdk',
+      'iphonesimulator',
+      '-configuration',
+      xcodeBuildConfiguration,
+      'SYMROOT=${iPhoneSimulatorDirectory.path}',
+      'ARCHS=x86_64',
+      'ONLY_ACTIVE_ARCH=NO' // No device targeted, so build all valid architectures.
+    ];
+
+    await processUtils.run(
+      pluginsBuildCommand,
+      workingDirectory: flutterProject.ios.hostAppRoot.childDirectory('Pods').path,
+      allowReentrantFlutter: false,
+    );
+
+    final Directory iPhoneBuiltProductsDirectory = iPhoneModeDirectory.childDirectory('$xcodeBuildConfiguration-iphoneos');
```

Comment:
Style nit - some of these lines and identifiers are really long. I would split up more work onto multiple lines, and use comments instead of long identifiers to clarify what directories are what.

Reply:
renamed